File: src/DomainServices/Services/UserService.cs
using System;
using System.Collections.Generic;
using System.IdentityModel.Tokens.Jwt;
using System.Linq;
using System.Security.Claims;
using System.Text;
using System.Threading.Tasks;
using IQuality.DomainServices.Interfaces;
using IQuality.Infrastructure.Database.Repositories.Interface;
using IQuality.Models;
using IQuality.Models.Authentication;
using IQuality.Models.Forms;
using IQuality.Models.Helpers;
using Microsoft.AspNetCore.Identity;
using Microsoft.Extensions.Configuration;
using Microsoft.IdentityModel.Tokens;
using Raven.Client.Documents.Linq;

namespace IQuality.DomainServices.Services
{
    [Injectable]
    public class UserService: IUserService
    {

        private IUserRepository _userRepository;
        public UserService(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }
        
        public async Task<List<ApplicationUser>> GetUsers()
        {
            return await _userRepository.GetAllWhereAsync((item) => item.Email != null);
        }

        public async Task<ApplicationUser> GetUser(string applicationUserId)
        {
            return await _userRepository.GetByIdAsync(applicationUserId);
        }
        
        public async Task DeactivateUser(string applicationUserId)
        {
            ApplicationUser user = await _userRepository.GetByIdAsync(applicationUserId);
            user.Active = false;
            await _userRepository.SaveAsync(user);
        }
        
        public async Task DeleteUser(string applicationUserId)
        {
            ApplicationUser user = await _userRepository.GetByIdAsync(applicationUserId);
            _userRepository.Delete(user);
        }
    }
}

File: src/Web/src/app/core/services/user.service.ts
import { Injectable } from '@angular/core';
import {ApiService} from "@IQuality/core/services/api.service";
import {ApplicationUser} from "@IQuality/core/models/application-user";
import {AuthenticationService} from "@IQuality/core/services/authentication.service";

@Injectable({
  providedIn: 'root'
})
export class UserService {

  public async firstTime(): Promise<boolean> {
    let user = await this.getUser(this.authService.getNameIdentifier);
    return user.firstTime;
  }

  private _user: ApplicationUser = null;

  constructor(private _api: ApiService, private authService: AuthenticationService) {
  }

  getUsers(){
    return this._api.get<Array<ApplicationUser>>('/users', {});
  }

  getUser(applicationUserId: string){
    return this._api.get<ApplicationUser>(`/users/${applicationUserId}`, {});
  }

  // This will prob. "deactivate" the account.
  public deactivateUser(applicationUserId: string){
    return this._api.put(`/users/${applicationUserId}`, {});
  }

  public deleteUser(applicationUserId: string){
    return this._api.delete(`/users/${applicationUserId}`);
  }
}
